src/components/SearchMap/SearchMapWithGoogleMaps.js

```diff
@@ -3,21 +3,29 @@ import ReactDOM from 'react-dom';
 import invariant from 'invariant';
 import { arrayOf, func, node, number, oneOfType, shape, string } from 'prop-types';
 import isEqual from 'lodash/isEqual';
-import { withGoogleMap, GoogleMap, OverlayView } from 'react-google-maps';
-import { OVERLAY_VIEW } from 'react-google-maps/lib/constants';
+import classNames from 'classnames';
 import { types as sdkTypes } from '../../util/sdkLoader';
 import { parse } from '../../util/urlHelpers';
 import { propTypes } from '../../util/types';
 import { ensureListing } from '../../util/data';
 import { sdkBoundsToFixedCoordinates, hasSameSDKBounds } from '../../util/maps';
+import { getOffsetOverride, getLayoutStyles } from '../../util/googleMaps';
 import { SearchMapInfoCard, SearchMapPriceLabel, SearchMapGroupLabel } from '../../components';
 
 import { groupedByCoordinates, reducedToArray } from './SearchMap.helpers.js';
+import css from './SearchMapWithGoogleMaps.module.css';
 
 export const LABEL_HANDLE = 'SearchMapLabel';
 export const INFO_CARD_HANDLE = 'SearchMapInfoCard';
 const BOUNDS_FIXED_PRECISION = 8;
 
+const OVERLAY_VIEW = 'overlayView';
+const FLOAT_PANE = 'floatPane';
+const MAP_PANE = 'mapPane';
+const MARKER_LAYER = 'markerLayer';
+const OVERLAY_LAYER = 'overlayLayer';
+const OVERLAY_MOUSE_TARGET = 'overlayMouseTarget';
```

Comment:
If these constants are values defined in the Google Maps API, you should clarify that very explicitly here. At least add a comment to the API reference, maybe even prefix the vars with `GMAPS_` etc?

Reply:
We use just 2 of them, so I left only those and added better code comments to provide some context what these are.

```js
// Panes on Google Maps specify the stacking order for different layers on the map.
// https://developers.google.com/maps/documentation/javascript/customoverlays#intitialize
//
// Google Maps uses 5 different panes:
// 'mapPane', 'overlayLayer', 'markerLayer', 'overlayMouseTarget', 'floatPane'
// We only need the last 2:
// - 'overlayMouseTarget': for the SearchMapPriceLabelWithOverlay & SearchMapGroupLabelWithOverlay.
// - 'floatPane': to render InfoCardComponent on top of price and group labels.
```